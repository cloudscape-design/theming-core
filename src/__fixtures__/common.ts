// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
import {
  Theme,
  Mode,
  Context,
  Override,
  ThemePreset,
  FullResolution,
  SpecificResolution,
  FullResolutionPaths,
} from '../shared/theme';

export function createStubPropertiesMap(theme: Theme): Record<string, string> {
  return Object.keys(theme.tokens).reduce((acc, curr) => {
    acc[curr] = `--${curr}-css`;
    return acc;
  }, {} as Record<string, string>);
}

export function createStubVariablesMap(theme: Theme): Record<string, string> {
  return Object.keys(theme.tokens).reduce((acc, curr) => {
    acc[curr] = `${curr}-var`;
    return acc;
  }, {} as Record<string, string>);
}

export const colorMode: Mode = {
  id: 'color',
  states: {
    light: { default: true },
    dark: { selector: '.dark', media: 'not print' },
  },
};

export const densityMode: Mode = {
  id: 'density',
  states: {
    compact: { selector: '.compact' },
    comfortable: { default: true },
  },
};

export const motionMode: Mode = {
  id: 'motion',
  states: {
    default: { default: true },
    disabled: { selector: '.disabled-motion' },
  },
};

export const navigationContext: Context = {
  id: 'navigation',
  selector: '.navigation',
  tokens: {
    shadow: {
      light: '{black}', // example of a reference
      dark: '{brown}',
    },
    boxShadow: {
      dark: 'purple', // example of a literal value
      light: 'purple',
    },
  },
};

const anotherContext: Context = {
  id: 'another-context',
  selector: '.another-context',
  tokens: {
    shadow: {
      light: '{black}', // example of a reference
      dark: '{brown}',
    },
    boxShadow: {
      dark: 'purple', // example of a literal value
      light: 'purple',
    },
  },
};

const emptyTheme: Theme = {
  id: 'root',
  selector: ':root',
  tokens: {},
  tokenModeMap: {},
  contexts: {},
  modes: {},
};

export const rootTheme: Theme = {
  id: 'root',
  selector: ':root',
  tokens: {
    fontFamilyBase: '"Helvetica Neue", Arial, sans-serif',
    fontFamilyBody: '{fontFamilyBase}',
    black: 'black',
    grey: 'grey',
    brown: 'brown',
    shadow: {
      light: '{grey}',
      dark: '{black}',
    },
    buttonShadow: `{shadow}`,
    boxShadow: {
      light: '{shadow}',
      dark: '{brown}',
    },
    lineShadow: {
      light: '{buttonShadow}',
      dark: '{boxShadow}',
    },
    small: '1px',
    medium: '3px',
    scaledSize: {
      compact: '{small}',
      comfortable: '{medium}',
    },
    appear: {
      default: '20ms',
      disabled: '0',
    },
    containerShadowBase: '2px 3px orange, -1px 0 8px olive',
    modalShadowContainer: '{containerShadowBase}',
  },
  tokenModeMap: {
    shadow: 'color',
    buttonShadow: 'color',
    boxShadow: 'color',
    lineShadow: 'color',
    scaledSize: 'density',
    appear: 'motion',
  },
  contexts: {
    navigation: navigationContext,
  },
  modes: {
    density: densityMode,
    color: colorMode,
    motion: motionMode,
  },
};

export const secondaryTheme: Theme = {
  ...rootTheme,
  id: 'secondary',
  selector: '.secondary-theme',
  tokens: {
    ...rootTheme.tokens,
    black: 'purple',
    brown: 'black',
  },
  contexts: {
    navigation: {
      ...navigationContext,
      tokens: {
        shadow: {
          light: '{black}',
          dark: '{grey}',
        },
      },
    },
  },
};

const anotherSecondaryTheme: Theme = {
  ...secondaryTheme,
  contexts: {
    navigation: {
      ...navigationContext,
    },
    'another-context': {
      ...anotherContext,
    },
  },
};

export const themesWithCircularDependencies: Array<Theme> = [
  {
    ...emptyTheme,
    tokens: {
      firstToken: '{firstToken}',
    },
  },
  {
    ...emptyTheme,
    tokens: {
      firstToken: '{secondToken}',
      secondToken: '{thirdToken}',
      thirdToken: '{secondToken}',
    },
  },
];

export const themeWithNonExistingToken: Theme = {
  ...emptyTheme,
  tokens: {
    firstToken: '{nonExistingToken}',
  },
};

export const fullResolution: FullResolution = {
  black: 'black',
  grey: 'grey',
  brown: 'brown',
  shadow: {
    light: 'grey',
    dark: 'black',
  },
  buttonShadow: {
    light: 'grey',
    dark: 'black',
  },
  boxShadow: {
    light: 'grey',
    dark: 'brown',
  },
  lineShadow: {
    light: 'grey',
    dark: 'brown',
  },
  small: '1px',
  medium: '3px',
  scaledSize: {
    compact: '1px',
    comfortable: '3px',
  },
  appear: {
    default: '20ms',
    disabled: '0',
  },
  containerShadowBase: '2px 3px orange, -1px 0 8px olive',
  modalShadowContainer: '2px 3px orange, -1px 0 8px olive',
  fontFamilyBase: '"Helvetica Neue", Arial, sans-serif',
  fontFamilyBody: '"Helvetica Neue", Arial, sans-serif',
};

export const fullResolutionPaths: FullResolutionPaths = {
  black: ['black'],
  grey: ['grey'],
  brown: ['brown'],
  shadow: {
    light: ['shadow', 'grey'],
    dark: ['shadow', 'black'],
  },
  buttonShadow: {
    light: ['buttonShadow', 'shadow', 'grey'],
    dark: ['buttonShadow', 'shadow', 'black'],
  },
  boxShadow: {
    light: ['boxShadow', 'shadow', 'grey'],
    dark: ['boxShadow', 'brown'],
  },
  lineShadow: {
    light: ['lineShadow', 'buttonShadow', 'shadow', 'grey'],
    dark: ['lineShadow', 'boxShadow', 'brown'],
  },
  small: ['small'],
  medium: ['medium'],
  scaledSize: {
    compact: ['scaledSize', 'small'],
    comfortable: ['scaledSize', 'medium'],
  },
  appear: {
    default: ['appear'],
    disabled: ['appear'],
  },
  containerShadowBase: ['containerShadowBase'],
  modalShadowContainer: ['modalShadowContainer', 'containerShadowBase'],
  fontFamilyBase: ['fontFamilyBase'],
  fontFamilyBody: ['fontFamilyBody', 'fontFamilyBase'],
};

export const defaultsResolution: SpecificResolution = {
  black: 'black',
  grey: 'grey',
  brown: 'brown',
  shadow: 'grey',
  buttonShadow: 'grey',
  boxShadow: 'grey',
  lineShadow: 'grey',
  small: '1px',
  medium: '3px',
  scaledSize: '3px',
  appear: '20ms',
  fontFamilyBase: '"Helvetica Neue", Arial, sans-serif',
  fontFamilyBody: '"Helvetica Neue", Arial, sans-serif',
  containerShadowBase: '2px 3px orange, -1px 0 8px olive',
  modalShadowContainer: '2px 3px orange, -1px 0 8px olive',
};

export const override: Override = {
  tokens: {
    shadow: {
      light: 'yellow',
      dark: 'orange',
    },
    boxShadow: {
      light: 'green',
    },
    lineShadow: {
      light: 'pink',
      dark: 'pink',
    },
    buttonShadow: 'red',
  },
  contexts: {
    navigation: {
      tokens: {
        shadow: {
          light: 'pink',
        },
        buttonShadow: {
          dark: 'green',
        },
      },
    },
  },
};

export const overrideWithRandomContext: Override = {
  tokens: { ...override.tokens },
  contexts: {
    randomContext: {
      tokens: {
        shadow: {
          light: 'pink',
        },
        buttonShadow: {
          dark: 'green',
        },
      },
    },
  },
};

export const preset: ThemePreset = {
  theme: rootTheme,
  themeable: ['shadow', 'boxShadow', 'buttonShadow', 'lineShadow'],
  exposed: ['shadow', 'buttonShadow', 'boxShadow', 'lineShadow'],
  propertiesMap: createStubPropertiesMap(rootTheme),
  variablesMap: createStubVariablesMap(rootTheme),
};

export const presetWithSecondaryTheme: ThemePreset = {
  ...preset,
  secondary: [secondaryTheme],
};
export const anotherPresetWithSecondaryTheme: ThemePreset = {
  ...preset,
  secondary: [anotherSecondaryTheme],
};
