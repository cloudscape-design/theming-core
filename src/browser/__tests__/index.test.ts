// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
import { afterEach, describe, test, expect } from 'vitest';
import { preset, presetWithSecondaryTheme, override } from '../../__fixtures__/common';
import { applyTheme } from '../index';

afterEach(() => {
  allStyleNodes().forEach((tag) => tag.remove());
});

describe('without secondary theme', () => {
  test('attaches one style node containing override', () => {
    applyTheme({ override, preset });

    const styleNodes = allStyleNodes();

    expect(styleNodes).toHaveLength(1);
    const themeNode = styleNodes[0];

    expect(themeNode.innerHTML).toMatchSnapshot();
  });

  test('removes style node on reset', () => {
    const { reset } = applyTheme({ override, preset });

    reset();

    expect(allStyleNodes()).toHaveLength(0);
  });
});

describe('with secondary theme', () => {
  test('attaches one style node containing override', () => {
    applyTheme({ override, preset: presetWithSecondaryTheme });

    const styleNodes = allStyleNodes();

    expect(styleNodes).toHaveLength(1);
    const themeNode = styleNodes[0];

    expect(themeNode.innerHTML).toMatchSnapshot();
  });

  test('removes style node on reset', () => {
    const { reset } = applyTheme({ override, preset: presetWithSecondaryTheme });

    reset();

    expect(allStyleNodes()).toHaveLength(0);
  });
});

describe('with baseThemeId', () => {
  test('attaches one style node containing overrides with the correct theme selector', () => {
    applyTheme({ override, preset: presetWithSecondaryTheme, baseThemeId: 'secondary' });

    const styleNodes = allStyleNodes();

    expect(styleNodes).toHaveLength(1);
    const themeNode = styleNodes[0];

    expect(themeNode.innerHTML).toMatchSnapshot();
  });

  test('throws error if baseThemeId is not available', () => {
    expect(() => applyTheme({ override, preset: presetWithSecondaryTheme, baseThemeId: 'invalid' })).toThrow(
      `Specified baseThemeId 'invalid' is not available. Available values are 'root', 'secondary'.`
    );
  });
});

const allStyleNodes = () => document.head.querySelectorAll('style');
