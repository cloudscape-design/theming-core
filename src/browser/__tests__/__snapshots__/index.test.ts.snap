// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with baseThemeId attaches one style node containing overrides with the correct theme selector 1`] = `
".secondary-theme.secondary-theme:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:purple;
	--grey-css:grey;
	--brown-css:black;
	--shadow-css:yellow;
	--buttonShadow-css:red;
	--boxShadow-css:green;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:3px;
	--appear-css:20ms;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.compact.compact.secondary-theme:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:purple;
	--grey-css:grey;
	--brown-css:black;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:1px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
@media not print {.dark.dark.secondary-theme:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:purple;
	--grey-css:grey;
	--brown-css:black;
	--shadow-css:orange;
	--buttonShadow-css:red;
	--boxShadow-css:black;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}}
.disabled-motion.disabled-motion.secondary-theme:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:purple;
	--grey-css:grey;
	--brown-css:black;
	--small-css:1px;
	--medium-css:3px;
	--appear-css:0;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.secondary-theme.secondary-theme .navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:purple;
	--grey-css:grey;
	--brown-css:black;
	--shadow-css:pink;
	--buttonShadow-css:red;
	--boxShadow-css:green;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:3px;
	--appear-css:20ms;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.navigation.navigation.secondary-theme:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:purple;
	--grey-css:grey;
	--brown-css:black;
	--shadow-css:pink;
	--buttonShadow-css:red;
	--boxShadow-css:green;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:3px;
	--appear-css:20ms;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.compact.compact.secondary-theme .navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:purple;
	--grey-css:grey;
	--brown-css:black;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:1px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.compact.compact.navigation.secondary-theme:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:purple;
	--grey-css:grey;
	--brown-css:black;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:1px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
@media not print {.dark.dark.secondary-theme .navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:purple;
	--grey-css:grey;
	--brown-css:black;
	--shadow-css:grey;
	--buttonShadow-css:green;
	--boxShadow-css:black;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}}
@media not print {.dark.dark.navigation.secondary-theme:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:purple;
	--grey-css:grey;
	--brown-css:black;
	--shadow-css:grey;
	--buttonShadow-css:green;
	--boxShadow-css:black;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}}
.disabled-motion.disabled-motion.secondary-theme .navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:purple;
	--grey-css:grey;
	--brown-css:black;
	--small-css:1px;
	--medium-css:3px;
	--appear-css:0;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.disabled-motion.disabled-motion.navigation.secondary-theme:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:purple;
	--grey-css:grey;
	--brown-css:black;
	--small-css:1px;
	--medium-css:3px;
	--appear-css:0;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}"
`;

exports[`with secondary theme attaches one style node containing override 1`] = `
":root:root{
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--shadow-css:yellow;
	--buttonShadow-css:red;
	--boxShadow-css:green;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:3px;
	--appear-css:20ms;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.compact.compact:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:1px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
@media not print {.dark.dark:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--shadow-css:orange;
	--buttonShadow-css:red;
	--boxShadow-css:brown;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}}
.disabled-motion.disabled-motion:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--small-css:1px;
	--medium-css:3px;
	--appear-css:0;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.navigation.navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--shadow-css:pink;
	--buttonShadow-css:red;
	--boxShadow-css:purple;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:3px;
	--appear-css:20ms;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.compact.compact .navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:1px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.compact.compact.navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:1px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
@media not print {.dark.dark .navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--shadow-css:brown;
	--buttonShadow-css:green;
	--boxShadow-css:purple;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}}
@media not print {.dark.dark.navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--shadow-css:brown;
	--buttonShadow-css:green;
	--boxShadow-css:purple;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}}
.disabled-motion.disabled-motion .navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--small-css:1px;
	--medium-css:3px;
	--appear-css:0;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.disabled-motion.disabled-motion.navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--small-css:1px;
	--medium-css:3px;
	--appear-css:0;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}"
`;

exports[`without secondary theme attaches one style node containing override 1`] = `
":root:root{
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--shadow-css:yellow;
	--buttonShadow-css:red;
	--boxShadow-css:green;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:3px;
	--appear-css:20ms;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.compact.compact:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:1px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
@media not print {.dark.dark:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--shadow-css:orange;
	--buttonShadow-css:red;
	--boxShadow-css:brown;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}}
.disabled-motion.disabled-motion:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--small-css:1px;
	--medium-css:3px;
	--appear-css:0;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.navigation.navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--shadow-css:pink;
	--buttonShadow-css:red;
	--boxShadow-css:purple;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:3px;
	--appear-css:20ms;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.compact.compact .navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:1px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.compact.compact.navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--small-css:1px;
	--medium-css:3px;
	--scaledSize-css:1px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
@media not print {.dark.dark .navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--shadow-css:brown;
	--buttonShadow-css:green;
	--boxShadow-css:purple;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}}
@media not print {.dark.dark.navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--shadow-css:brown;
	--buttonShadow-css:green;
	--boxShadow-css:purple;
	--lineShadow-css:pink;
	--small-css:1px;
	--medium-css:3px;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}}
.disabled-motion.disabled-motion .navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--small-css:1px;
	--medium-css:3px;
	--appear-css:0;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}
.disabled-motion.disabled-motion.navigation:not(#\\\\9){
	--fontFamilyBase-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--fontFamilyBody-css:\\"Helvetica Neue\\", Arial, sans-serif;
	--black-css:black;
	--grey-css:grey;
	--brown-css:brown;
	--small-css:1px;
	--medium-css:3px;
	--appear-css:0;
	--containerShadowBase-css:2px 3px orange, -1px 0 8px olive;
	--modalShadowContainer-css:2px 3px orange, -1px 0 8px olive;
}"
`;
